packages/concerto-core/test/serializer.js

```diff
@@ -232,6 +243,60 @@ describe('Serializer', () => {
             });
         });
 
+        it('should generate concept with a map value', () => {
+            let address = factory.newConcept('org.acme.sample', 'Address');
+            address.city = 'Winchester';
+            address.country = 'UK';
+            address.elevation = 3.14;
+            address.postcode = 'SO21 2JN';
+            address.dict = {
+                $class: 'org.acme.sample.Dictionary',
+                value: new Map(Object.entries({'Lorem':'Ipsum'}))
```

Comment:
Yeah, this looks pretty ugly to me, compared to supporting JS objects here... rather than forcing people to use the Map type.

Reply:
I think that there's a crossed wire here...

We absolutely must support deserialization from vanilla JSON.

However, this test is creating a Resource instance with an internal representation of the map. We could argue that that is not a good design, but that's separate from de(serialization) IMO.